test/spec/modules/appnexusBidAdapter_spec.js

```diff
@@ -632,6 +653,28 @@ describe('AppNexusAdapter', function () {
         'value': ['rock', 'pop']
       }, {
         'key': 'test'
+      }, {
+        'key': 'tools',
+        'value': ['power', 'industrial', 'home']
+      }, {
+        'key': 'power tools'
+      }, {
+        'key': 'drills'
+      }, {
+        'key': 'video'
+      }, {
+        'key': 'source',
+        'value': ['streaming']
+      }, {
+        'key': 'renting'
+      }, {
+        'key': 'app',
+        'value': ['iphone 11']
+      }, {
+        'key': 'appcontent',
+        'value': ['home repair']
+      }, {
+        'key': 'dyi'
```

Comment:
would you consider gathering this https://github.com/prebid/Prebid.js/blob/d73944600b1e15a5a1f5c4ce5c2db8a84eb53644/modules/permutiveRtdProvider.js#L231 and https://github.com/prebid/Prebid.js/blob/d73944600b1e15a5a1f5c4ce5c2db8a84eb53644/modules/1plusXRtdProvider.js#L108 and https://github.com/prebid/Prebid.js/blob/d73944600b1e15a5a1f5c4ce5c2db8a84eb53644/modules/airgridRtdProvider.js#L76 in scope for this change?

Reply:
these rtd modules bypass the "normal" workflow that the data controller module is able to edit

Reply:
If they're populating the values under the bid.params.keywords before our request goes out, it should be gathered automatically by the logic we have already?  I'm not sure how else to pre-emptively gather the data from their systems.

If they want to change their setup to have the publisher populate the ortb2 keyword fields with their data, then I suppose their customizations shouldn't be needed (if that's what you meant by 'in scope for this change').

Reply:
These rtd modules typically write contextual segments to site.content.data and user segments to user.data. It is in those locations that the data controller module is able to edit the segments down if necessary. We believe, since the appnexus adapter does not support either of these inputs and people seem to be using params.keywords in their place, that appnexus adapter should be responsible for copying the segments from the standard locations to the keywords instead of these modules each having special appnexus handling.

Reply:
as reference: https://github.com/prebid/Prebid.js/pull/8614#pullrequestreview-1035918465

Reply:
How would you transform the typical segment objects into key-value pairs (I'm looking at the samples in the (first party data)[https://docs.prebid.org/features/firstPartyData.html]) page from the docs, and I don't see exactly how things should be pulled together)?  

Is this really something that would okay to implement for everyone who potentially has this data in these fields, as opposed to just doing it for these rtd modules?

Reply:
I agree it isn't obvious how this should be done, and each of these adapters seems to have made slightly different choices to support Xandr

Reply:
If someone change suggest a standard approach, we can take a look and consider.  If it's going to take time, then maybe it can be a future PR and we just have this PR focus on the keywords fields specifically.